apache/metron-bro-plugin-kafka
# METRON-1329: Simplify metron-bro-plugin-kafka package loading

This is a very simple change that normalizes the process of loading this package in an existing bro system.  This **must** be merged in coordination with [METRON-1313](https://github.com/apache/metron/issues/847) against apache/metron.  The reason why is because HEAD [currently points to](https://github.com/apache/metron/blob/8022f2c8c4e9018a15a4f04d0a66f8bc0ea653c3/metron-deployment/roles/bro/tasks/metron-bro-plugin-kafka.yml#L40) `Bro/Kafka/logs-to-kafka.bro`, but after this change that would need to point to `Apache/Kafka/logs-to-kafka.bro`.  However, with the METRON-1313 changes, we begin using `bro-pkg` which is a much cleaner method of install/upgrade.

This also assumes an appropriately set 0.1 git tag on apache/metron-bro-plugin-kafka.

# Testing
1.  Spin up a centos 7 VM.
    ```
    vagrant init bento/centos-7.4
    vagrant up
    vagrant ssh
    sudo su -
    ```
1.  [Install bro 2.5.2](https://www.bro.org/sphinx/install/install.html) and [bro-pkg](http://bro-package-manager.readthedocs.io/en/stable/quickstart.html#installation).  Make sure you are running at least bro 2.5 and bro-pkg 1.2.0, and configure bro-pkg properly.
    ```
    yum -y install epel-release
    yum -y install cmake make gcc gcc-c++ flex bison libpcap-devel openssl-devel python-devel swig zlib-devel git python-pip librdkafka
    git clone --recursive git://git.bro.org/bro
    cd bro
    git checkout v2.5.2
    git submodule update --recursive
    ./configure && make && make install
    export PATH=$PATH:/usr/local/bro/bin
    pip install bro-pkg
    bro-pkg autoconfig
    wget https://github.com/edenhill/librdkafka/archive/v0.9.4.tar.gz -O /tmp/v0.9.4.tar.gz
    cd /tmp
    tar -xvf v0.9.4.tar.gz
    cd librdkafka-0.9.4
    ./configure --prefix=/usr/local && make && make install
    ```
1.  Install the bro package, and load it
    ```
    bro-pkg install https://github.com/jonzeolla/metron-bro-plugin-kafka --force --version 0.1
    echo "redef Kafka::logs_to_send = set(HTTP::LOG, DNS::LOG);" >> /usr/local/bro/share/bro/site/local.bro
    ```
1.  Run a `broctl deploy`.
    ```
    sed -i 's/eth0/enp0s3/' /usr/local/bro/etc/node.cfg
    broctl deploy
    ```

If this was successful, the last step should look like this:
```
[root@node1 site]# broctl deploy
checking configurations ...
installing ...
creating policy directories ...
installing site policies ...
generating standalone-layout.bro ...
generating local-networks.bro ...
generating broctl-config.bro ...
generating broctl-config.sh ...
stopping ...
bro not running
starting ...
starting bro ...
```

Otherwise, it will return an error.
